### PR TITLE
feat(tests): port TouchToEmptyAccountRevert legacy tests

### LINKED ISSUE #2174: Add tests for "reverted touch of empty account"
The scenario of revering a touch of an empty account is completely missing from tests (including static).

Consider porting "TouchToEmptyAccountRevert*.json" from legacytests.

### PR DESCRIPTION
**Pull request by chfast**
## 🗒️ Description
- Port 6 legacy `TouchToEmptyAccountRevert` static tests for EIP-161 empty account cleanup
- 3 test functions × 2 parametrized target balance variants (empty/funded):
  - `test_touch_empty_account_call_chain`: zero-value CALL through a call chain
  - `test_touch_empty_account_call_then_oog`: direct CALL + OOG sub-call
  - `test_touch_empty_account_selfdestruct_then_oog`: SELFDESTRUCT touch + OOG sub-call
- Valid from Istanbul to London (EIP-161 applies, pre-EIP-1559)

## 🔗 Related Issues or PRs
Closes #2174

## ✅ Checklist

- [x] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx tox -e static
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-specs/blob/HEAD/docs/) directory.
- [ ] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.
- [ ] Tests: For PRs implementing a missed test case, update the [post-mortem document](/ethereum/execution-specs/blob/HEAD/docs/writing_tests/post_mortems.md) to add an entry the list.
- [ ] Ported Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) or [tests/static](/ethereum/execution-specs/blob/HEAD/tests/static) have been assigned `@ported_from` marker.

#### Cute Animal Picture

:worm:

### CODE REVIEW

**Comment by felix314159** [comment]
Ready for review or still in draft?

**Comment by chfast** [comment]
> Ready for review or still in draft?

Ready now.

File: tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+) -> None:
+    """
+    Test EIP-161 deletion of an account touched by zero-value CALL
+    through a call chain.
+
+    Call chain: tx -> entry -> intermediary -> target.
+    An empty target is deleted; a funded target survives.
+    """
+    storage = Storage()
+    target = pre.empty_account()
```
Comment (marioevz) [inline tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py]:
```suggestion
    target = pre.fund_eoa(amount=target_balance)
```
`pre.fund_eoa(amount=0)` is equivalent to `pre.empty_account()`, so it should be ok for both cases.

Comment (marioevz) [inline tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py:67]: I'm now thinking that this `empty_account` name could be misleading. We are not placing an account with all fields set to zero in the pre-state, as one might expect an empty account to be, rather the returned address is an address that is nowhere in the pre-state.

Perhaps `pre.nonexistent_address()` is a better name.

I don't think we currently can put an actual empty account in the pre-state, but after looking at the original static tests, it doesn't try to put an empty account either, so it might be ok?

File: tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+    if target_balance > 0:
+        pre[target] = Account(balance=target_balance)
```
Comment (marioevz) [inline tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py]: Need to add `pytest.mark.pre_alloc_mutable` due to this line.

**Review by marioevz — COMMENTED** [review]
Some comments, thanks!

**Comment by marioevz** [comment]
Another comment is that these files should be deleted in this PR:
```
tests/static/state_tests/stRevertTest/TouchToEmptyAccountRevert2_ParisFiller.json
tests/static/state_tests/stRevertTest/TouchToEmptyAccountRevert3_ParisFiller.json
tests/static/state_tests/stRevertTest/TouchToEmptyAccountRevert_ParisFiller.json
```

File: tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py (hunk 1 above)
Comment (chfast) [reply tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py:67]: It looks I moved too fast with these tests, sorry. The main point I need precise this scenario which is not covered in evmone by EEST: there is an empty account in the pre-state, it is touched during execution, then this touch is reverted and the account is not deleted in the end.

I identified only this legacy test (from legacytests repo) coverts it. Now I understand why: this test is filled for Paris+ only in EEST.

Can I create completely empty account in the pre-state pre-Paris?

Comment (marioevz) [reply tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py:67]: So, afaik:
- The test framework has a check to disallow empty accounts in the pre-allocation (but we can workaround this depending on the next point)
- I'm not sure if the same check exists on the EELS side in one or multiple forks, perhaps @SamWilsn or @gurukamath know
- If not possible, this might be one reason why we would not be able to delete legacy-tests, cc @danceratopz

Comment (chfast) [reply tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py:67]: I will investigate this.

**Comment by felix314159** [comment]
please rebase this so it can be reviewed again & merged, given that #2462 is merged u will have to replace `pre.empty_account()` with `pre.nonexistent_account()`.

File: tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py (hunk 1 above)
Comment (felix314159) [reply tests/spurious_dragon/eip161_empty_account_cleanup/test_touch_empty_account_revert.py]: result of investigation?
